rest/data/host.go

```diff
@@ -165,3 +168,62 @@ func FindHostByIdWithOwner(hostID string, user gimlet.User) (*host.Host, error)
 
 	return hostById, nil
 }
+
+var errHostStatusChangeConflict = errors.New("conflicting host status modification is in progress")
+
+// TerminateSpawnHost enqueues a job to terminate a spawn host.
+func TerminateSpawnHost(ctx context.Context, env evergreen.Environment, u *user.DBUser, h *host.Host) (int, error) {
+	if h.Status == evergreen.HostTerminated {
+		return http.StatusBadRequest, errors.Errorf("host '%s' is already terminated", h.Id)
+	}
+
+	ts := utility.RoundPartOfMinute(1).Format(units.TSFormat)
+	terminateJob := units.NewSpawnHostTerminationJob(h, u.Id, ts)
+	if err := env.RemoteQueue().Put(ctx, terminateJob); err != nil {
+		if amboy.IsDuplicateJobScopeError(err) {
+			err = errHostStatusChangeConflict
+		}
+		return http.StatusInternalServerError, err
+	}
+
+	return http.StatusOK, nil
+}
+
+// StopSpawnHost enqueues a job to stop a running spawn host.
+func StopSpawnHost(ctx context.Context, env evergreen.Environment, u *user.DBUser, h *host.Host) (int, error) {
+	if h.Status == evergreen.HostStopped {
+		return http.StatusBadRequest, errors.Errorf("host '%s' is already stopped", h.Id)
+	}
+	if h.Status != evergreen.HostRunning && h.Status != evergreen.HostStopping {
```

Comment:
Hmm yeah, it seems like Spruce should also allow stopping hosts to be stopped again.

Alternatively, we could get rid of the "stopping" intermediate state altogether since it only serves as a convenience warning to the user.

Reply:
One way or another works for me. I mean, I like the stopping state as a user, but if it has to go 🤷 .

Reply:
I filed a UI ticket ([EVG-16712](https://jira.mongodb.org/browse/EVG-16712)) to let users re-press the stop button if something goes wrong.